File: storm-spring/src/test/java/st/orm/spring/TestRepositoryBeanFactoryPostProcessor.java
package st.orm.spring;

import org.springframework.context.annotation.Configuration;

/**
 * BeanFactoryPostProcessor that scans base packages for Repository interfaces and registers them as beans.
 */
@Configuration
public class TestRepositoryBeanFactoryPostProcessor extends RepositoryBeanFactoryPostProcessor {
    @Override
    public String getOrmTemplateBeanName() {
        return "ormTemplate";
    }

    @Override
    public String[] getRepositoryBasePackages() {
        return new String[] { "st.orm.spring.repository" };
    }
}

File: storm-kotlin-spring/src/test/kotlin/st/orm/spring/TestRepositoryBeanFactoryPostProcessor.kt
package st.orm.spring

import org.springframework.context.annotation.Configuration
import st.orm.spring.kotlin.RepositoryBeanFactoryPostProcessor

@Configuration
internal open class TestRepositoryBeanFactoryPostProcessor : RepositoryBeanFactoryPostProcessor() {
    override fun getOrmTemplateBeanName(): String = "ormTemplate"

    // Make platform wide repositories available as well in the context of the dataORMTemplate.
    override fun getRepositoryBasePackages(): Array<String> = arrayOf("st.orm.spring.repository")
}
